From d645a18bd5b51ad8a9b63d7e7016f64e4907092b Mon Sep 17 00:00:00 2001
From: Rob Herring <r.herring@freescale.com>
Date: Fri, 1 Aug 2008 15:24:36 -0500
Subject: [PATCH] ENGR00086352 Clean-up PMIC kconfig and print statements

Clean-up the PMIC Kconfig options and makefiles
Remove extra MC13892 print statements

Signed-off-by: Rob Herring <r.herring@freescale.com>
---
 arch/arm/configs/imx27ads_defconfig     |   11 +---
 arch/arm/configs/imx31_3stack_defconfig |    7 +--
 arch/arm/configs/imx31ads_defconfig     |   10 ++--
 arch/arm/configs/imx35_3stack_defconfig |    9 ++--
 arch/arm/configs/imx37_3stack_defconfig |    6 +-
 arch/arm/configs/imx51_3stack_defconfig |    8 +--
 drivers/mxc/pmic/Kconfig                |   89 ++++++++-----------------------
 drivers/mxc/pmic/core/Makefile          |   27 ++-------
 drivers/mxc/pmic/core/pmic_core_i2c.c   |   11 +---
 drivers/regulator/Kconfig               |   32 ++++++-----
 10 files changed, 68 insertions(+), 142 deletions(-)

diff --git a/arch/arm/configs/imx27ads_defconfig b/arch/arm/configs/imx27ads_defconfig
index d3e2024..f1a4ac5 100644
--- a/arch/arm/configs/imx27ads_defconfig
+++ b/arch/arm/configs/imx27ads_defconfig
@@ -426,12 +426,6 @@ CONFIG_FW_LOADER=m
 # CONFIG_SYS_HYPERVISOR is not set
 CONFIG_CONNECTOR=y
 CONFIG_PROC_EVENTS=y
-
-#
-# Power Management IC's
-#
-# CONFIG_PMIC is not set
-# CONFIG_PMIC_WM8350 is not set
 CONFIG_MTD=y
 # CONFIG_MTD_DEBUG is not set
 # CONFIG_MTD_CONCAT is not set
@@ -1382,14 +1376,15 @@ CONFIG_MXC_DAM=y
 #
 # MXC PMIC support
 #
-CONFIG_MXC_SPI_PMIC_CORE=y
 CONFIG_MXC_PMIC=y
+CONFIG_MXC_PMIC_MC13783=y
+# CONFIG_MXC_PMIC_MC13892 is not set
+# CONFIG_MXC_PMIC_MC9SDZ60 is not set
 CONFIG_MXC_PMIC_CHARDEV=y
 
 #
 # MXC PMIC Client Drivers
 #
-CONFIG_MXC_PMIC_MC13783=y
 CONFIG_MXC_MC13783_ADC=y
 CONFIG_MXC_MC13783_AUDIO=y
 CONFIG_MXC_MC13783_RTC=y
diff --git a/arch/arm/configs/imx31_3stack_defconfig b/arch/arm/configs/imx31_3stack_defconfig
index bb5d99a..6680f20 100644
--- a/arch/arm/configs/imx31_3stack_defconfig
+++ b/arch/arm/configs/imx31_3stack_defconfig
@@ -1369,16 +1369,15 @@ CONFIG_MXC_DAM=y
 #
 # MXC PMIC support
 #
-CONFIG_MXC_SPI_PMIC_CORE=y
-# CONFIG_MXC_I2C_MCU_PMIC_CORE is not set
-# CONFIG_MXC_I2C_PMIC_CORE is not set
 CONFIG_MXC_PMIC=y
+CONFIG_MXC_PMIC_MC13783=y
+# CONFIG_MXC_PMIC_MC13892 is not set
+# CONFIG_MXC_PMIC_MC9SDZ60 is not set
 CONFIG_MXC_PMIC_CHARDEV=y
 
 #
 # MXC PMIC Client Drivers
 #
-CONFIG_MXC_PMIC_MC13783=y
 CONFIG_MXC_MC13783_ADC=y
 CONFIG_MXC_MC13783_AUDIO=y
 CONFIG_MXC_MC13783_RTC=y
diff --git a/arch/arm/configs/imx31ads_defconfig b/arch/arm/configs/imx31ads_defconfig
index c12a47b..ae506d9 100644
--- a/arch/arm/configs/imx31ads_defconfig
+++ b/arch/arm/configs/imx31ads_defconfig
@@ -558,10 +558,9 @@ CONFIG_MTD_NAND_MXC_V2=y
 #
 CONFIG_REGULATOR_API=y
 CONFIG_REGULATOR=y
+# CONFIG_REGULATOR_DEBUG is not set
 CONFIG_REGULATOR_MC13783=y
-
-#
-#
+# CONFIG_REGULATOR_WM8350 is not set
 # CONFIG_PARPORT is not set
 # CONFIG_PNP is not set
 CONFIG_BLK_DEV=y
@@ -1616,14 +1615,15 @@ CONFIG_MXC_DAM=y
 #
 # MXC PMIC support
 #
-CONFIG_MXC_SPI_PMIC_CORE=y
 CONFIG_MXC_PMIC=y
+CONFIG_MXC_PMIC_MC13783=y
+# CONFIG_MXC_PMIC_MC13892 is not set
+# CONFIG_MXC_PMIC_MC9SDZ60 is not set
 CONFIG_MXC_PMIC_CHARDEV=y
 
 #
 # MXC PMIC Client Drivers
 #
-CONFIG_MXC_PMIC_MC13783=y
 CONFIG_MXC_MC13783_ADC=y
 CONFIG_MXC_MC13783_AUDIO=y
 CONFIG_MXC_MC13783_RTC=y
diff --git a/arch/arm/configs/imx35_3stack_defconfig b/arch/arm/configs/imx35_3stack_defconfig
index 0302d51..5d9017d 100644
--- a/arch/arm/configs/imx35_3stack_defconfig
+++ b/arch/arm/configs/imx35_3stack_defconfig
@@ -494,8 +494,8 @@ CONFIG_MTD_NAND_MXC_V2=y
 CONFIG_REGULATOR_API=y
 CONFIG_REGULATOR=y
 # CONFIG_REGULATOR_DEBUG is not set
-# CONFIG_REGULATOR_WM8350 is not set
 CONFIG_REGULATOR_MAX8660=y
+# CONFIG_REGULATOR_WM8350 is not set
 # CONFIG_PARPORT is not set
 # CONFIG_PNP is not set
 CONFIG_BLK_DEV=y
@@ -1275,16 +1275,15 @@ CONFIG_MXC_IPU_PF=y
 #
 # MXC PMIC support
 #
-# CONFIG_MXC_SPI_PMIC_CORE is not set
-CONFIG_MXC_I2C_MCU_PMIC_CORE=y
-# CONFIG_MXC_I2C_PMIC_CORE is not set
 CONFIG_MXC_PMIC=y
+# CONFIG_MXC_PMIC_MC13783 is not set
+# CONFIG_MXC_PMIC_MC13892 is not set
+CONFIG_MXC_PMIC_MC9SDZ60=y
 CONFIG_MXC_PMIC_CHARDEV=y
 
 #
 # MXC PMIC Client Drivers
 #
-CONFIG_MXC_PMIC_MC9SDZ60=y
 # CONFIG_MXC_MC9SDZ60_ADC is not set
 CONFIG_MXC_MC9SDZ60_RTC=y
 CONFIG_MXC_MC9SDZ60_POWER=y
diff --git a/arch/arm/configs/imx37_3stack_defconfig b/arch/arm/configs/imx37_3stack_defconfig
index 373b9aa..208d8a9 100644
--- a/arch/arm/configs/imx37_3stack_defconfig
+++ b/arch/arm/configs/imx37_3stack_defconfig
@@ -1281,9 +1281,9 @@ CONFIG_MXC_IPU_V3=y
 #
 # MXC PMIC support
 #
-# CONFIG_MXC_SPI_PMIC_CORE is not set
-# CONFIG_MXC_I2C_MCU_PMIC_CORE is not set
-# CONFIG_MXC_PMIC is not set
+# CONFIG_MXC_PMIC_MC13783 is not set
+# CONFIG_MXC_PMIC_MC13892 is not set
+# CONFIG_MXC_PMIC_MC9SDZ60 is not set
 
 #
 # Advanced Power Management devices
diff --git a/arch/arm/configs/imx51_3stack_defconfig b/arch/arm/configs/imx51_3stack_defconfig
index f725a31..18131ce 100644
--- a/arch/arm/configs/imx51_3stack_defconfig
+++ b/arch/arm/configs/imx51_3stack_defconfig
@@ -1,7 +1,6 @@
 #
 # Automatically generated make config: don't edit
 # Linux kernel version: 2.6.24
-# Wed Jul 30 17:04:37 2008
 #
 CONFIG_ARM=y
 CONFIG_SYS_SUPPORTS_APM_EMULATION=y
@@ -853,16 +852,15 @@ CONFIG_RTC_LIB=y
 #
 # MXC PMIC support
 #
-# CONFIG_MXC_SPI_PMIC_CORE is not set
-# CONFIG_MXC_I2C_MCU_PMIC_CORE is not set
-CONFIG_MXC_I2C_PMIC_CORE=y
 CONFIG_MXC_PMIC=y
+# CONFIG_MXC_PMIC_MC13783 is not set
+CONFIG_MXC_PMIC_MC13892=y
+# CONFIG_MXC_PMIC_MC9SDZ60 is not set
 # CONFIG_MXC_PMIC_CHARDEV is not set
 
 #
 # MXC PMIC Client Drivers
 #
-CONFIG_MXC_PMIC_MC13892=y
 CONFIG_MXC_MC13892_ADC=y
 CONFIG_MXC_MC13892_RTC=y
 CONFIG_MXC_MC13892_LIGHT=y
diff --git a/drivers/mxc/pmic/Kconfig b/drivers/mxc/pmic/Kconfig
index 8b20bf3..4da37d3 100644
--- a/drivers/mxc/pmic/Kconfig
+++ b/drivers/mxc/pmic/Kconfig
@@ -4,32 +4,31 @@
 
 menu "MXC PMIC support"
 
-config MXC_SPI_PMIC_CORE
-	tristate "PMIC Protocol support (SPI interface)"
-	depends on ARCH_MXC && SPI_MXC
-	default n
+config MXC_PMIC
+	boolean
+
+config MXC_PMIC_MC13783
+	tristate "MC13783 PMIC"
+	depends on ARCH_MXC && SPI
+	select MXC_PMIC
 	---help---
-	This is the PMIC core/protocol driver for the Freescale MXC application.
-	SPI should be providing the interface between the PMIC and the MCU.
-	You must select the SPI driver support to enable this option.
+	  This is the MXC MC13783(PMIC) support. It include
+	  ADC, Audio, Battery, Connectivity, Light, Power and RTC.
 
-config MXC_I2C_MCU_PMIC_CORE
-        tristate "MCU PMIC Protocol support (I2C interface)"
-        depends on ARCH_MXC
-        default n
-        ---help---
-        This is the PMIC core/protocol driver for the MX35 3DS MCU PMIC
+config MXC_PMIC_MC13892
+	tristate "MC13892 PMIC"
+	depends on ARCH_MXC && I2C
+	select MXC_PMIC
+	---help---
+	  This is the MXC MC13892(PMIC) support. It include
+	  ADC, Battery, Connectivity, Light, Power and RTC.
 
-config MXC_I2C_PMIC_CORE
-        tristate "PMIC Protocol support (I2C interface)"
-        depends on ARCH_MXC
-        default n
-        ---help---
-        This is the PMIC core/protocol driver for the MX51 3DS PMIC
-	
-config MXC_PMIC
-	boolean
-	default (MXC_SPI_PMIC_CORE || MXC_I2C_MCU_PMIC_CORE || MXC_I2C_PMIC_CORE)
+config MXC_PMIC_MC9SDZ60
+	tristate "MC9sDZ60 PMIC"
+	depends on ARCH_MXC && I2C
+	select MXC_PMIC
+	  ---help---
+	  This is the MXC MC9sDZ60(MCU) PMIC support.
 
 config MXC_PMIC_CHARDEV
 	tristate "MXC PMIC device interface"
@@ -43,53 +42,9 @@ config MXC_PMIC_CHARDEV
 comment "MXC PMIC Client Drivers"
 	depends on MXC_PMIC
 
-config MXC_PMIC_MC13783
-	tristate "MC13783 Client Drivers"
-	depends on MXC_SPI_PMIC_CORE
-	default n
-	---help---
-	This is the MXC MC13783(PMIC) client drivers support. It include
-	ADC, Audio, Battery, Connectivity, Light, Power and RTC.
-
 source "drivers/mxc/pmic/mc13783/Kconfig"
 
-config MXC_PMIC_MC13892
-	tristate "MC13892 Client Drivers"
-	depends on MXC_I2C_PMIC_CORE
-	default n
-	---help---
-	This is the MXC MC13892(PMIC) client drivers support. It include
-	ADC, Battery, Connectivity, Light, Power and RTC.
-
 source "drivers/mxc/pmic/mc13892/Kconfig"
-
-config MXC_PMIC_SC55112
-	tristate "SC55112 Client Drivers"
-	depends on MXC_SPI_PMIC_CORE && ARCH_MXC91131
-	default n
-	---help---
-	This is the MXC MC13783(PMIC) client drivers support. It include
-	ADC, Audio, Battery, Connectivity, Light, Power and RTC.
-
-config MXC_PMIC_FIXARB
-        bool "Use Arbitration Bit Workaround"
-        depends on MXC_PMIC_SC55112
-        default n
-        ---help---
-            Enable this option to include the code that will automatically
-            toggle the sc55112 arbitration bits to allow register write
-            access to the secondary processor. Note that this also requires
-            a hardware modification to the RF Deck that connects the Sphinx
-            card (with the sc55112 PMIC) to the MXC91131 EVB.
-
-config MXC_PMIC_MC9SDZ60
-        tristate "MC9sDZ60 Client Drivers"
-        depends on MXC_I2C_MCU_PMIC_CORE
-        depends on !ARCH_MXC91131
-        default n
-        ---help---
-        This is the MXC MC9sDZ60(MCU) PMIC client drivers support.
-
 source "drivers/mxc/pmic/mc9sdz60/Kconfig"
 
 endmenu
diff --git a/drivers/mxc/pmic/core/Makefile b/drivers/mxc/pmic/core/Makefile
index 57e5612..b7768c7 100644
--- a/drivers/mxc/pmic/core/Makefile
+++ b/drivers/mxc/pmic/core/Makefile
@@ -1,29 +1,14 @@
 #
 # Makefile for the PMIC core drivers.
 #
-obj-$(CONFIG_MXC_SPI_PMIC_CORE) += pmic_core_spi_mod.o
+obj-$(CONFIG_MXC_PMIC_MC13783) += pmic_mc13783_mod.o
+pmic_mc13783_mod-objs := pmic_external.o pmic_event.o pmic_core_spi.o mc13783.o
 
-obj-$(CONFIG_MXC_I2C_MCU_PMIC_CORE) += pmic_core_i2c_mod.o
+obj-$(CONFIG_MXC_PMIC_MC9SDZ60) += pmic_mc9sdz60_mod.o
+pmic_mc9sdz60_mod-objs := pmic_external.o pmic_event.o mcu_pmic_core.o max8660.o mc9sdz60.o
 
-obj-$(CONFIG_MXC_I2C_PMIC_CORE) += pmic_core_mc13892_mod.o
-
-pmic_core_mc13892_mod-objs := pmic_external.o pmic_event.o pmic_core_i2c.o mc13892.o
-
-pmic_core_i2c_mod-objs := pmic_external.o pmic_event.o mcu_pmic_core.o
-
-pmic_core_spi_mod-objs := pmic_external.o pmic_event.o pmic_core_spi.o
+obj-$(CONFIG_MXC_PMIC_MC13892) += pmic_mc13892_mod.o
+pmic_mc13892_mod-objs := pmic_external.o pmic_event.o pmic_core_i2c.o mc13892.o
 
 obj-$(CONFIG_MXC_PMIC_CHARDEV)	+= pmic-dev.o
 
-ifeq ($(CONFIG_MXC_PMIC_MC13783),y)
-pmic_core_spi_mod-objs += mc13783.o
-endif
-
-ifeq ($(CONFIG_MXC_PMIC_SC55112),y)
-pmic_core_spi_mod-objs += sc55112.o
-endif
-
-ifeq ($(CONFIG_MXC_PMIC_MC9SDZ60), y)
-pmic_core_i2c_mod-objs += mc9sdz60.o
-pmic_core_i2c_mod-objs += max8660.o
-endif
diff --git a/drivers/mxc/pmic/core/pmic_core_i2c.c b/drivers/mxc/pmic/core/pmic_core_i2c.c
index 919f4e3..0e1a5fc 100644
--- a/drivers/mxc/pmic/core/pmic_core_i2c.c
+++ b/drivers/mxc/pmic/core/pmic_core_i2c.c
@@ -158,7 +158,6 @@ static int __devinit is_chip_onboard(struct i2c_client *client)
 
 	/*bind the right device to the driver */
 	ret = pmic_i2c_24bit_read(client, REG_IDENTIFICATION);
-	dev_err(&client->dev, "identification %x\n", ret);
 
 	if (MC13892_GEN_ID_VALUE != BITFEXT(ret, MC13892_GENERATION_ID)) {
 		/*compare the address value */
@@ -168,7 +167,6 @@ static int __devinit is_chip_onboard(struct i2c_client *client)
 			MC13892_GEN_ID_VALUE);
 		return -1;
 	}
-	dev_err(&client->dev, "chip on board!\n");
 
 	return 0;
 }
@@ -238,8 +236,6 @@ static int __devinit pmic_probe(struct i2c_client *client)
 	int ret = 0;
 	int pmic_irq;
 
-	pr_debug("start probe pmic 13892\n");
-
 	ret = is_chip_onboard(client);
 
 	if (ret == -1)
@@ -283,7 +279,7 @@ static int __devinit pmic_probe(struct i2c_client *client)
 
 	pmic_pdev_register();
 
-	pr_debug("Device %s probed\n", client->dev.bus_id);
+	dev_info(&client->dev, "Loaded\n");
 
 	return PMIC_SUCCESS;
 }
@@ -320,20 +316,17 @@ static struct i2c_driver pmic_driver = {
 
 static int __init pmic_init(void)
 {
-	pr_debug("Registering the PMIC Protocol Driver\n");
 	return i2c_add_driver(&pmic_driver);
 }
 
 static void __exit pmic_exit(void)
 {
-	pr_debug("Unregistering the PMIC Protocol Driver\n");
-	return i2c_del_driver(&pmic_driver);
+	i2c_del_driver(&pmic_driver);
 }
 
 /*
  * Module entry points
  */
-/* subsys_initcall_sync(pmic_init); */
 subsys_initcall_sync(pmic_init);
 module_exit(pmic_exit);
 
diff --git a/drivers/regulator/Kconfig b/drivers/regulator/Kconfig
index 7c3de61..c9bc775 100644
--- a/drivers/regulator/Kconfig
+++ b/drivers/regulator/Kconfig
@@ -18,19 +18,26 @@ config REGULATOR_DEBUG
 	  Say yes here to enable debugging support.
 
 config REGULATOR_MC13783
-    tristate "MC13783 Regulator Support"
-        depends on REGULATOR
-        depends on MXC_PMIC_MC13783
+	tristate "MC13783 Regulator Support"
+	depends on REGULATOR
+	depends on MXC_PMIC_MC13783
+	default y
 
 config REGULATOR_MC13892
-    tristate "MC13892 Regulator Support"
-        depends on REGULATOR
-        depends on MXC_PMIC_MC13892
-	
+	tristate "MC13892 Regulator Support"
+	depends on REGULATOR
+	depends on MXC_PMIC_MC13892
+	default y
+
+config REGULATOR_MAX8660
+	tristate "MAX8660 Regulator Support"
+	depends on REGULATOR
+	depends on MXC_PMIC_MC9SDZ60
+	default y
+
 config REGULATOR_WM8350
-    tristate "WM8350 Regulator Support"
-        depends on REGULATOR
-        depends on !MXC_PMIC_MC13783
+	tristate "WM8350 Regulator Support"
+	depends on REGULATOR
 
 menu "WM8350 Config Mode"
 depends on REGULATOR_WM8350
@@ -55,9 +62,4 @@ endchoice
 
 endmenu
 
-config REGULATOR_MAX8660
-    tristate "MAX8660 Regulator Support"
-       depends on REGULATOR
-       depends on MXC_PMIC_MC9SDZ60
-
 endmenu
-- 
1.5.4.4

